### PR TITLE
Optimize Dockerfile

### LINKED ISSUE #13: Optimize Dockerfile

### PR DESCRIPTION
**Pull request by MakMuftic**
Optimize Dockerfile PR:
- replaced make commands with native go commands
- final image only contains binary (uses `scratch` image as a starting point)

closes #13

### CODE REVIEW

**Review by mpetrunic — COMMENTED** [review]
Could you double check image is using non root user?

File: Dockerfile [hunk 1]
```diff
@@ -1,11 +1,35 @@
-FROM golang:alpine
+FROM golang:alpine AS builder
 
-RUN mkdir /app
-ADD . /app/
-WORKDIR /app
+# Set necessary environmet variables needed for our image
```
Comment (mpetrunic) [inline Dockerfile]: Typo in environment

File: Dockerfile [hunk 2]
```diff
@@ -1,11 +1,35 @@
...
+COPY . .
+
+# Build the application
+RUN go build -o vedran .
+
+# Move to /dist directory as the place for resulting binary folder
+WORKDIR /dist
+
+# Copy binary from build to main folder
+RUN cp /build/vedran .
```
Comment (mpetrunic) [inline Dockerfile]: You can just copy from build in final stage

**Comment by MakMuftic** [comment]
> Could you double check image is using non root user?

Do you mean for this final image that is built from `scratch` image?

**Comment by mpetrunic** [comment]
> > Could you double check image is using non root user?
> 
> Do you mean for this final image that is built from `scratch` image? 

Yes

**Comment by MakMuftic** [comment]
This is ready for merge.

**Review by mpetrunic — APPROVED** [review]

**Review by mpetrun5 — APPROVED** [review]

**Pull request merged**